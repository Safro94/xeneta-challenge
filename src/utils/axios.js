import axios from 'axios';
import https from 'https';

const opts = {
	httpsAgent: new https.Agent({ rejectUnauthorized: false }),
	baseURL: process.env.REACT_APP_SERVER_URL,
};

const interceptorError = error => Promise.reject(error.response);

const instance = axios.create(opts);

instance.interceptors.response.use(response => response, interceptorError);
instance.defaults.headers.common['x-api-key'] = process.env.REACT_APP_API_KEY;

export default instance;
