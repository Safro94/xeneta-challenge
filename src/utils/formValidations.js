const validateSearchForm = ({
	departure,
	destination,
	departureDate,
	returnDate,
}) => {
	if (!departure || !destination || !departureDate) {
		return false;
	}

	if (departure.code === destination.code) {
		return false;
	}

	if (returnDate && returnDate < departureDate) {
		return false;
	}

	return true;
};

const errorCompare = (trip, message) => {
	const { departure, destination } = trip;

	const status = !departure && !destination ? true : false;

	return {
		message,
		status: !status && departure?.code === destination?.code,
	};
};

export { validateSearchForm, errorCompare };
