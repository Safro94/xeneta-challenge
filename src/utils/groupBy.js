import { marketPositions } from 'constants/marketPositions';

const groupBy = data =>
	data.reduce((accum, item) => {
		if (!accum[marketPositions.low]) accum[marketPositions.low] = [];
		if (!accum[marketPositions.average]) accum[marketPositions.average] = [];
		if (!accum[marketPositions.high]) accum[marketPositions.high] = [];

		accum[marketPositions.low].push({ x: item.day, y: item.low });
		accum[marketPositions.average].push({ x: item.day, y: item.mean });
		accum[marketPositions.high].push({ x: item.day, y: item.high });

		return accum;
	}, {});

export default groupBy;
