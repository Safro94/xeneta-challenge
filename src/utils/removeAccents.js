const removeAccents = value => {
	const map = {
		a: 'á|à|ã|â',
		e: 'é|è|ê',
		i: 'í|ì|î',
		o: 'ó|ò|ô|õ',
		u: 'ú|ù|û|ü',
	};

	for (let pattern in map) {
		value = value.replace(new RegExp(map[pattern], 'g'), pattern);
	}

	return value;
};

export { removeAccents };
