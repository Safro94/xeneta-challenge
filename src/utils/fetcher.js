import axios from 'utils/axios';

const fetcher = async ({ method = 'get', url, data = null }) => {
	return axios[method](url, JSON.parse(data)).then(
		response => response.data,
		error => {
			throw error;
		}
	);
};

export default fetcher;
