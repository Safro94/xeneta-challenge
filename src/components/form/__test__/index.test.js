import { render, screen } from 'utils/test-utils';
import Form from '../';

jest.mock('react-nice-dates', () => ({
	DateRangePicker: () => <div>DateRangePicker</div>,
}));

describe('Form', () => {
	it('should match the snapshot', () => {
		const { container } = render(
			<Form>
				<Form.Autocomplete
					placeholder='Departure'
					value='2020-05-01'
					source={[]}
					onSelectItem={() => {}}
					onBlurSelected
					id='departure'
				/>

				<Form.Datepicker onSelectItem={() => {}} />
				<Form.Submit disabled>Search</Form.Submit>
			</Form>
		);

		expect(screen.getByPlaceholderText(/Departure/i)).toBeInTheDocument();
		expect(screen.getByText(/DateRangePicker/i)).toBeInTheDocument();
		expect(screen.getByText(/search/i)).toBeDisabled();
		expect(container.firstElementChild).toMatchSnapshot();
	});

	it('should have a button not disabled', () => {
		render(
			<Form>
				<Form.Datepicker onSelectItem={() => {}} />
				<Form.Submit>Search</Form.Submit>
			</Form>
		);

		expect(screen.getByText(/search/i)).not.toBeDisabled();
	});
});
