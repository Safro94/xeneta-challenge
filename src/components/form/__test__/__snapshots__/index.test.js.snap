// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form should match the snapshot 1`] = `
<form>
  <div
    class="formGroup"
  >
    <div
      aria-expanded="false"
      aria-haspopup="listbox"
      aria-owns="react-autowhatever-1"
      class="container"
      role="combobox"
    >
      <div>
        <input
          aria-autocomplete="list"
          aria-controls="react-autowhatever-1"
          autocomplete="off"
          class="input"
          data-testid="input-text"
          placeholder="Departure"
          type="text"
          value=""
        />
      </div>
      <div
        id="react-autowhatever-1"
        role="listbox"
      />
    </div>
  </div>
  <div
    class="formGroup"
  >
    <div>
      DateRangePicker
    </div>
  </div>
  <div
    class="formGroup"
  >
    <button
      class="button"
      disabled=""
    >
      Search
    </button>
  </div>
</form>
`;
